### PR TITLE
Added API routes for Admins to GetAllUsers and to ban users

### LINKED ISSUE #352: As an admin I want to be able to access a list of users on the website
## Why not in capstone?
At the beginning of SOEN 490 the team voted for the most important features to implement. Due to lack of time, the admin panel was ruled as unnecessary

## Acceptance Criteria:  
- [ ] API route returns a paginated list of users on the website

### LINKED ISSUE #353: As an admin, I want to be able to BAN users, so that they can never access the website again
## Why not in capstone?
At the beginning of SOEN 490 the team voted for the most important features to implement. Due to lack of time, the admin panel was ruled as unnecessary

## Acceptance Criteria:  
- [ ] Only admins can ban users
- [ ] Users banned cannot login anymore and their podcasts are hidden

### PR DESCRIPTION
**Pull request by shadijiha**
Banned users are globally filtered from all queries therefore they cannot login/access website

Closes #352 
Closes #353

### CODE REVIEW

File: Backend/Backend.Tests/AdminPanelTests.cs
Comment (BigSaladMan) [inline Backend/Backend.Tests/AdminPanelTests.cs:10]: Im going to reword this test suite tomorrow, wait till then to merge this

Comment (BigSaladMan) [reply Backend/Backend.Tests/AdminPanelTests.cs:10]: Structure doesnt follow the others

File: Backend/Backend/Controllers/AdminPanelController.cs
Comment (BigSaladMan) [inline Backend/Backend/Controllers/AdminPanelController.cs:23]: You forgot to add the logger

File: Backend/Backend/Controllers/AdminPanelController.cs
Comment (BigSaladMan) [inline Backend/Backend/Controllers/AdminPanelController.cs:24]: You should prob add a check to see if the user is logged in.

Comment (BigSaladMan) [reply Backend/Backend/Controllers/AdminPanelController.cs:24]: Something that might be interesting to do on this controller, is to add to the service a version of the authService.IdentifyUserAsync where you do the same thing as that one, but check to see if its an admin at the same time.
For Reference 
image

File: Backend/Backend/Controllers/AdminPanelController.cs
Comment (BigSaladMan) [inline Backend/Backend/Controllers/AdminPanelController.cs:36]: Error message is a bit off. This error happens when you are not logged in.

**Review by BigSaladMan — APPROVED** [review]
Dont Merge this yet if you have the req amount, wait till im finished reworking the test suite tmr beforehand
Edit: Done

File: Backend/Backend/Controllers/AdminPanelController.cs
Comment (shadijiha) [reply Backend/Backend/Controllers/AdminPanelController.cs:24]: This already done with the middleware called "VerifyAdmin"

File: Backend/Backend/Controllers/AdminPanelController.cs
Comment (shadijiha) [reply Backend/Backend/Controllers/AdminPanelController.cs]: This should never happen since the route is protected by a middleware. We need error message to be debuggable

File: Backend/Backend.Tests/AdminPanelTests.cs
Comment (BigSaladMan) [reply Backend/Backend.Tests/AdminPanelTests.cs:20]: All done

**Review by mubashir494 — APPROVED** [review]

**Pull request merged**
